test/ng/compileSpec.js

```diff
@@ -3128,17 +3128,129 @@ describe('$compile', function() {
       });
     });
 
-    it('should observe interpolated attrs', inject(function($rootScope, $compile) {
-      $compile('<div some-attr="{{value}}" observer></div>')($rootScope);
+    describe('$compileProvider.restrictDirectivesTo', function() {
 
-      // should be async
-      expect(observeSpy).not.toHaveBeenCalled();
+      var collected;
+      beforeEach(module(function($compileProvider) {
+        collected = false;
+        $compileProvider.directive('testCollect', function() {
+          return {
+            restrict: 'ECMA',
+            link: function(scope, element) {
+              collected = true;
+            }
+          };
+        });
+      }));
 
-      $rootScope.$apply(function() {
-        $rootScope.value = 'bound-value';
+      describe('EA', function() {
+
+        beforeEach(module(function($compileProvider) {
+          $compileProvider.restrictDirectivesTo('EA');
+        }));
+
+        var $compile, $rootScope;
+        beforeEach(inject(function(_$compile_,_$rootScope_) {
+          $compile = _$compile_;
+          $rootScope = _$rootScope_;
+        }));
+
+        it('should not compile class directives', function() {
```

Comment:
I think it would be cleaner to use `test-collect` for all testcases. It makes the comparison easier.